api/envoy/config/cluster/v3/outlier_detection.proto

```diff
@@ -9,17 +9,46 @@ import "udpa/annotations/status.proto";
 import "udpa/annotations/versioning.proto";
 import "validate/validate.proto";
 
+import "envoy/type/v3/range.proto";
+
 option java_package = "io.envoyproxy.envoy.config.cluster.v3";
 option java_outer_classname = "OutlierDetectionProto";
 option java_multiple_files = true;
 option go_package = "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3;clusterv3";
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Outlier detection]
+// Error bucket for HTTP codes
+// [#not-implemented-hide:]
+message HttpErrorsBucket {
+  string name = 1;
+  type.v3.Int32Range range = 2;
+}
+
+// Error bucket for locally originated errors
+// [#not-implemented-hide:]
+message LocalOriginEvents {
+}
+
+// Error bucket for database errors.
+// Sub-parameters may be added later, like malformed response, error on write, etc.
+// [#not-implemented-hide:]
+message DatabaseTransactions {
+}
+
+// Union of possible error buckets.
+// [#not-implemented-hide:]
+message ErrorBucket {
+  oneof bucket {
```

Comment:
We now prefer not to use `oneof` in the xDS API, as per https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md.